examples/stream-hello-world/StreamHelloWorld_Client.cpp

```diff
@@ -26,20 +28,22 @@ int main(int argc, char* argv[]) {
   {
     // this example runs inside the Future.then lambda
     LOG(INFO) << "------------------ Run in future.then";
-    auto s = std::make_shared<ExampleSubscriber>(5, 6);
+    auto s = yarpl::Reference<yarpl::Subscriber<Payload>>(
+        new ExampleSubscriber(5, 6));
     rsf->connect().then([s](std::shared_ptr<RSocketRequester> rs) {
-      rs->requestStream(Payload("Bob"), s);
+      rs->requestStream(Payload("Bob"))->subscribe(s);
     });
-    s->awaitTerminalEvent();
+    //    s->awaitTerminalEvent();
```

Comment:
No, your understanding is correct: In this case, you have "s", which you can hang onto, and invoke methods on. It is an additional reference.